src/snappy_device_agents/devices/muxpi/muxpi.py

```diff
@@ -236,13 +236,15 @@ def get_image_type(self):
         def check_path(dir):
             self._run_control("test -e {}".format(dir))
 
-        # First check if this is a limerick image
+        # First check if this is a ce-oem-iot image
         try:
             disk_info_path = self.mount_point / "writable/.disk/info"
-            self._run_control(f"grep limerick {disk_info_path}")
-            return "limerick"
+            buildstamp = '"iot-[a-z]+-[a-z-]*(classic-(server|desktop)-[0-9]+'
+            buildstamp += '|core-[0-9]+)"'
```

Comment:
this is really clever and I like it! This way we can still work with limerick, erlangen, etc. However, although it's easy to understand now while it's fresh in our minds, it's also going to get forgotten and could easily be broken later.

device agent code is notoriously difficult to write unit tests for, because so much of it talks to the physical device. We've kinda glossed over that in the past, but I'm trying to make some progress here and this looks like a great opportunity to add one. Can we break this bit out into a function (it can be in the same file) and create a unit test for it? the unit test should go under the muxpi dir. See the ones we have in the maas2 device agent as an example if you want.